### PR TITLE
Use selection for blob certificates

### PR DESCRIPTION
**Pull request by danielSanchezQ**
We really need to select the certificates not the blobs. I refactored everything to fit this. This will help in integrating @Zeegomo work on the certificates mempool.

### CODE REVIEW

File: nomos-core/src/da/certificate/mod.rs [hunk 1]
```diff
@@ -1 +1,22 @@
...
+}
+
+pub trait BlobCertificateSelect {
+    type Certificate: Certificate;
+    type Settings: Clone;
+
+    fn new(settings: Self::Settings) -> Self;
+    fn select_blob_from<'i, I: Iterator<Item = Self::Certificate> + 'i>(
+        &self,
+        blobs: I,
```
Comment (Zeegomo) [inline nomos-core/src/da/certificate/mod.rs:20]: shouldn't this be certificates?

File: nomos-da/full-replication/src/lib.rs [hunk 2]
```diff
@@ -123,12 +124,31 @@ impl attestation::Attestation for Attestation {
...
+}
+
+impl certificate::Certificate for Certificate {
+    type Blob = Blob;
+
+    fn blob(&self) -> <Self::Blob as blob::Blob>::Hash {
+        self.attestations[0].blob
+    }
+
+    fn as_bytes(&self) -> Bytes {
```
Comment (Zeegomo) [inline nomos-da/full-replication/src/lib.rs:145]: I supposed we should be able to deserialize from `as_bytes` but this is variable length without mentioning how many attestations we have, am I missing something? Also, wasn't as bytes supposed to be `wire::deserialize`?

File: nomos-core/src/da/certificate/mod.rs (hunk 1 above)
Comment (danielSanchezQ) [reply nomos-core/src/da/certificate/mod.rs:20]: yes!

File: nomos-da/full-replication/src/lib.rs (hunk 2 above)
Comment (danielSanchezQ) [reply nomos-da/full-replication/src/lib.rs:145]: Yes, but I saw it was manually added in the `Attestation` and did the same. Will change both (?)

File: nomos-core/src/block/builder.rs [hunk 3]
```diff
@@ -90,13 +91,13 @@ where
     }
 
     #[must_use]
-    pub fn with_blobs(mut self, blobs: impl Iterator<Item = B> + 'static) -> Self {
+    pub fn with_blobs(mut self, blobs: impl Iterator<Item = C> + 'static) -> Self {
```
Comment (youngjoon-lee) [inline nomos-core/src/block/builder.rs:94]: how about naming this as `with_blob_certificates` for clarity? also the name of parameter.

Comment (danielSanchezQ) [reply nomos-core/src/block/builder.rs:94]: Yes, missed it! Good catch!

**Review by youngjoon-lee — APPROVED** [review]
Looks good to me!

**Review by Zeegomo — APPROVED** [review]

**Pull request merged**